runcoms/zprofile

```diff
@@ -66,11 +66,11 @@ fi
 # Temporary Files
 #
 #
-if [[ -z "$TMPDIR" ]]; then
-  export TMPDIR="/tmp/zsh-$UID"
+# Set TMPDIR if the variable is not set or the directory doesn't exist
+if [[ ! "${TMPDIR}" ]]; then
```

Comment:
Why doesn't -z work? That should check if the string is zero length, which for an undefined or empty variable, it should be.